### PR TITLE
FEDX-51 : Updated null safety

### PR DESCRIPTION
**Pull request by tusharmathur-wk**
Migrate to null safety. Run the null safety migration tool with Dart migrate and check/adjust its suggestions if needed.
Apply the migration, and fix any static errors. Do a dart format, run the tests, fixing any issues.

### CODE REVIEW

**Comment by aviary3-wk** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

**Comment by semveraudit-wf** [comment]
### Public API Changes

 No changes to the public API found for commit 44fa98ba031e54b5d3045922c3f41361ed3d7d72

Showing results for 44fa98ba031e54b5d3045922c3f41361ed3d7d72
> Powered by [semver-audit-service](https://github.com/Workiva/semver-audit-service). Please report any problems by [filing an issue](https://github.com/Workiva/semver-audit-service/issues).
> Reported by the dart semver audit client 2.2.2
[Browse public API](https://semver-audit.wk-dev.wdesk.org/Workiva/contextual_message/44fa98ba031e54b5d3045922c3f41361ed3d7d72).

File: pubspec.yaml [hunk 1]
```diff
@@ -18,3 +18,9 @@ dev_dependencies:
   dependency_validator: '>=2.0.3 <4.0.0'
   test: ^1.16.5
   workiva_analysis_options: ^1.3.0
+
+dependency_overrides:
+  dart_dev_workiva:
+    git:
+      url: ssh://git@github.com/Workiva/dart_dev_workiva.git
+      ref: FEDX-41
```
Comment (robbecker-wf) [inline pubspec.yaml:26]: yay we can remove this now

File: lib/src/contextual_message.dart [hunk 2]
```diff
@@ -18,7 +18,7 @@ class ContextualMessage {
   String message;
 
   /// Optional context map, used for including additional data.
-  Map<String, dynamic> context;
+  Map<String, dynamic>? context;
```
Comment (robbecker-wf) [inline lib/src/contextual_message.dart:21]: 👍 this makes sense to be nullable

File: pubspec.yaml [hunk 3]
```diff
@@ -5,7 +5,7 @@ documentation: https://github.com/Workiva/contextual_message
 homepage: https://github.com/Workiva/contextual_message
 
 environment:
-  sdk: '>=2.11.0 <3.0.0'
+  sdk: '>=2.12.0 <3.0.0'
 
 dev_dependencies:
   dart_dev: '>=3.9.0 <5.0.0'
```
Comment (robbecker-wf) [inline pubspec.yaml:11]:
```suggestion
  dart_dev: ^4.0.0
```
Let's ensure we resolve to the null safe version of dart_dev

File: pubspec.yaml [hunk 4]
```diff
@@ -18,3 +18,9 @@ dev_dependencies:
   dependency_validator: '>=2.0.3 <4.0.0'
```
Comment (robbecker-wf) [inline pubspec.yaml:18]: We should also update the minimum of dart_style to ^2.0.0 and dependency_validator to ^3.0.0

File: lib/src/contextual_message.dart (hunk 2 above)
Comment (robbecker-wf) [inline lib/src/contextual_message.dart:21]:
```suggestion
  Map<String, dynamic> context;
```
I think we can make this non-nullable with a default in the constructor

File: lib/src/contextual_message.dart [hunk 5]
```diff
@@ -18,7 +18,7 @@ class ContextualMessage {
...
 
   /// Returns instance of ContextualMessage.
   ContextualMessage(this.message, {this.context});
```
Comment (robbecker-wf) [inline lib/src/contextual_message.dart:24]:
```suggestion
  ContextualMessage(this.message, {this.context = const {}});
```

File: test/contextual_message_test.dart [hunk 6]
```diff
@@ -6,7 +6,7 @@ void main() {
   test('ContextualMessage', () {
     final message = ContextualMessage('mathematics', context: {'2+2': '4'});
     expect(message.message, 'mathematics');
-    expect(message.context.containsKey('2+2'), isTrue);
-    expect(message.context['2+2'], '4');
+    expect(message.context?.containsKey('2+2'), isTrue);
+    expect(message.context?['2+2'], '4');
```
Comment (robbecker-wf) [inline test/contextual_message_test.dart:10]:
```suggestion
    expect(message.context.containsKey('2+2'), isTrue);
    expect(message.context['2+2'], '4');
```
We can get rid of these `?` if the other suggestions are accepted

**Review by robbecker-wf — APPROVED** [review]

**Comment by robbecker-wf** [comment]
+10 @Workiva/release-management-p

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM

**Pull request merged**